### DISCUSSION TITLE
Add Trustlist to API

### DISCUSSION THREAD

**Discussion by MichaelCasaDev** [GriefPrevention/GriefPrevention#1212] [Q&A] [answered]
Hi, i'm developing a plugin integrated with GriefPrevention, i need to check if a player is trusted in a claim, but i found nothing into the API Documentation!

This feature can be added to the new version of GriefPrevention? It's a simple function to be made public from private ;)

**Comment by Jikoo** [comment, marked as answer]
You're probably looking for `Claim#allowBuild` etc.

**Reply by MichaelCasaDev** [reply]
In server i can do /Trustlist when i'm in a claim
But from API i can't see who is in a trustlist

ex.
```
Griefprevention.getClaims() <- return a StringList of claims
Griefprevention.trustList(claimID) <- return a StringList of trusted players
```

**Reply by Jikoo** [reply]
Ah, you want `Claim#getPermissions(ArrayList, ArrayList, ArrayList, ArrayList)`

**Reply by MichaelCasaDev** [reply]
> Ah, you want `Claim#getPermissions(ArrayList, ArrayList, ArrayList, ArrayList)`

A method so I can fetch a claim by location (xyz) and if there is a claim check if a player object is trusted else nothing

**Reply by Jikoo** [reply]
You can get claims by location from `Datastore`. Again, really sounds like you're looking for `Claim#allow<X>` - if the player has the appropriate trust level, the various allow methods return null. If the player does not have permission to perform the operation, the denial message is returned.

**Reply by MichaelCasaDev** [reply]
>  `Datastore`. `Claim#allow<X>`
How i can do that in my code?

i have that code, how i can make it?

```
@EventHandler
public void onBlockBreak(BlockBreakEvent event){
    //Send message when player break a block in a claim that isn't him!!!
}
```

**Reply by Jikoo** [reply]
Have you tried looking at how GP handles block breaks? Your snippet (or lack thereof) is basically what GP already does - cancel the event, send a message.

Either way, you can access the `DataStore` instance from the exposed field `datastore` in the `GriefPrevention` instance. From there, you can call `DataStore#getClaimAt` to obtain the claim, if any, at a location.
I.e.:
```Java
GriefPrevention.instance.datastore.getClaimAt(location, ignoreHeight, cachedClaim)
```

**Comment by MichaelCasaDev** [comment]
Can mark that as resolved ;)

**Reply by RoboMWM** [reply]
the question asker can't mark an answer as the answer?
